### DISCUSSION TITLE
FetchError: Hostname/IP does not match certificate's altnames

### DISCUSSION THREAD

**Discussion by Matheart** [community/community#127829] [Copilot Conversations]
### Select Topic Area

Bug

### Body

When I try to use copilot in VS Code inside the server, this error message pops out
```
[INFO] [default] [2024-06-08T17:24:24.940Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-codex
[ERROR] [default] [2024-06-08T17:24:25.224Z] Request Error: Hostname/IP does not match certificate's altnames: Host: copilot-proxy.githubusercontent.com. is not in the cert's altnames: DNS:*.gist.githubusercontent.com, DNS:gist.githubusercontent.com
[ERROR] [ghostText] [2024-06-08T17:24:25.225Z] Error on ghost text request FetchError: Hostname/IP does not match certificate's altnames: Host: copilot-proxy.githubusercontent.com. is not in the cert's altnames: DNS:*.gist.githubusercontent.com, DNS:gist.githubusercontent.com
``` 
And copilot cannot be used normally, what should I do to fix this?

**Comment by Akash1134** [comment]
Hi @Matheart , The error message you're seeing suggests that there's a certificate mismatch issue with the proxy network. Mostly these errors are caused by a corporate proxy setup that uses custom certificates to intercept and inspect secure connections.

Please check this documentation: [Troubleshoot certificate related errors](https://docs.github.com/en/copilot/troubleshooting-github-copilot/troubleshooting-network-errors-for-github-copilot#troubleshooting-certificate-related-errors)

Hope that helps.
